Palindrome_permutation.py

```diff
@@ -1,25 +1,34 @@
-def is_permutation_of_palindrome(x):
-    # remove spaces and convert strings to lowercase
-    x = x.replace(" ", "").lower()
-
-    char_count = {}
-
-    for i in x:
-        if i.isalpha():
-            if i in char_count:
-                char_count[i] += 1
-            else:
-                char_count[i] = 1
-
-    odd_count = 0
-
-    for j in char_count.values():
-        if j % 2 != 0:
-            odd_count += 1
-
-    return odd_count <= 1
-
-a = str(input())
-print(is_permutation_of_palindrome(a))
+# In this problem we check if it's possible to obtain a mirrored string by simply rearranging the letters. "
+# If so, return - True, if not, return - False
+class Solution:
+    def __init__(self, str1):
+        self.str1 = str1
+
+    def is_permutation_of_palindrome(self):
+        # remove spaces and convert strings to lowercase

```

Comment:
Before running the entire logic to check if the string is palindrome, you can optimize the code by verifying the length of both strings same. I would encourage to check the book solution explained before submitting the pull request. I would assume book must have also noted this optimization.